eth/filters/api.go

```diff
@@ -345,6 +347,9 @@ func (api *PublicFilterAPI) GetLogs(ctx context.Context, crit FilterCriteria) ([
 		if crit.ToBlock != nil {
 			end = crit.ToBlock.Int64()
 		}
+		if end-begin+1 > api.maxBlockRange {
```

Comment:
This won't work if `begin` or `end` use `rpc.LatestBlockNumber`, `rpc.EarliestBlockNumber`, etc.

Reply:
If `begin == rpc.EarliestBlockNumber` and `end == rpc.LatestBlockNumber`, then the actual range can be huge but this check will pass.